python/fusion_engine_client/analysis/data_loader.py

```diff
@@ -309,6 +314,7 @@ def _read(self,
             'return_bytes': return_bytes,
             'return_message_index': return_message_index,
             'remove_nan_times': remove_nan_times,
+            'source_ids': source_ids,
```

Comment:
Convert to  set so order won't matter. Otherwise, if the user does `[0, 3]` one time and then `[3, 0]` another time the lists won't be equal and the cached data will be cleared.